### ISSUE TITLE
rust-libp2p still uses the term 'swarm'

### ISSUE THREAD

**Issue by tcharding** [libp2p/rust-libp2p#1647]
libp2p has moved to using the term 'switch' in place of the term 'swarm'. It is confusing for users of `rust-libp2p` to have different terminology or to have to write '... swarm (referred to as switch by the rest of libp2p) ...' when discussing code.

Has the topic of migrating `rust-libp2p` to use the term 'switch' been discussed already?

**Comment by tomaka** [comment]
During the initial implementation I used the term "swarm" with the sole reason of being consistent with the rest of the libp2p ecosystem. It's only much later that I learned that it had since then been renamed to "switch".

The problem now is that this is a high-effort low-reward change.
We would have to rename everything, fix all the documentation, publish a new crate (`libp2p-switch`).
While I agree that everything that happens on the wire should be precisely spec'ed and consistent between all libp2p implementations, I don't really understand why this should necessarily be the case for the API.
